### PR TITLE
CP Subsystem - Atomic Long

### PR DESCRIPTION
**Pull request by mdumandag**
depends on #222 

This PR adds CP-related codecs, AtomicLong implementation, documentation, and tests.

It also makes `Future#continue_with` support continuations that return Future by chaining them. It will be useful while implementing other proxies.

Protocol PR: https://github.com/hazelcast/hazelcast-client-protocol/pull/346

### CODE REVIEW

**Comment by mdumandag** [comment]
verify

**Review by puzpuzpuz — APPROVED** [review]
LGTM. Left a few nits.

File: hazelcast/proxy/cp/__init__.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        self._invocation_service = context.invocation_service
+        serialization_service = context.serialization_service
+        self._to_data = serialization_service.to_data
+        self._to_object = serialization_service.to_object
+
+    def destroy(self):
+        """Destroys this proxy.
+
+        Returns:
+            bool: ``True`` if this proxy is destroyed successfully, ``False`` otherwise.
```
[the remark this answers is not in the document]
Comment (mdumandag) [reply hazelcast/proxy/cp/__init__.py]: Good catch, in fact, we were returning ``None`` here, but the documentation was wrong.

File: hazelcast/cp.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            hazelcast.proxy.cp.atomic_long.AtomicLong: The AtomicLong proxy
+                for the given name.
+        """
+        return self._proxy_manager.get_or_create(ATOMIC_LONG_SERVICE, name)
+
+
+_DEFAULT_GROUP_NAME = "default"
+
+
+def _with_default_group_name(name):
```
[the remark this answers is not in the document]
Comment (mdumandag) [reply hazelcast/cp.py]: Changed it

File: hazelcast/cp.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+class CPProxyManager(object):
+    def __init__(self, context):
+        self._context = context
+
+    def get_or_create(self, service_name, proxy_name):
+        proxy_name = _with_default_group_name(proxy_name)
+        object_name = _get_object_name_for_proxy(proxy_name)
+
+        group_id = self._get_group_id(proxy_name)
+        if service_name == ATOMIC_LONG_SERVICE:
```
[the remark this answers is not in the document]
Comment (mdumandag) [reply hazelcast/cp.py:97]: I planned to do that in the last PR of the CP subsystem. Here it is, https://github.com/hazelcast/hazelcast-python-client/pull/227/commits/155d7bf47cdedf85ded1fe52de23a320cbad35f9#diff-02b2ff2fc57113af18673d95ac35fa6ae4286a25ddebf4599a8a765b0a594503R204

**Comment by mdumandag** [comment]
Thanks a lot for the review Andrey, I am merging it

File: hazelcast/cp.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply hazelcast/cp.py:97]: Thanks, makes sense.

**Pull request merged**